packages/auth/lib/Users.js

```diff
@@ -172,12 +179,39 @@ const signIn = async (_email, context, pgdb, req, consents, _tokenType) => {
     const init = await initiateSession({ req, pgdb, email, consents })
     const { country, phrase, session } = init
 
-    const token = await generateNewToken(tokenType, {
+    let authorizeToken = null
+
+    // Check {accessToken} and if is valid, try to generate a token in {authorizeToken}
+    if (accessToken) {
+      const accessTokenUser = await getUserByAccessToken(accessToken, { pgdb })
+
+      // Check if scope has authorizeSession prop and requesting user matches accessToken user
+      if (hasAuthorizeSession(accessTokenUser) && user && user.id === accessTokenUser.id) {
+        try {
+          authorizeToken = await generateNewToken(AUTHORIZE_TOKEN, {
+            pgdb,
+            session,
+            email,
+            accessToken,
+            context
+          })
+          tokenType = AUTHORIZE_TOKEN
+        } catch (e) {
+          // Fail silently
+          debug(e.message)
+        }
+      }
+    }
+
+    // Either user obtained token in {authorizeToken}, or generate a new
+    // token (because {authorizeSessionToken} is empty)
+    const token = authorizeToken || await generateNewToken(tokenType, {
```

Comment:
This is where the restriction, that `tokenType: AUTHORIZE_TOKEN` can't be specified to the `signIn` mutation, is this right? I really don't like the restriction, as it hinders me to specify a totaly legitimate mutation. `AUTHORIZE_TOKEN` is what I want, but as usual with `signIn` I am happy if it switches to something else. Please remove the restriction.